src/main/ApplicationImpl.cpp

```diff
@@ -568,6 +568,10 @@ ApplicationImpl::~ApplicationImpl()
     LOG_INFO(DEFAULT_LOG, "Application destructing");
     try
     {
+        if (mOverlayManager)
```

Comment:
the application doesn't always shutdown via `gracefulStop` (for example, if there's an exception). I added this so that we properly reset trackers to not call half-destroyed objects in `done` during shutdown. However, now that we call `done` manually, we don't really this special handling, so I think it can be removed.